### PR TITLE
Add EH support for CFGWalker

### PR DESCRIPTION
**Pull request by aheejin**
This adds EH instruction support for `CFGWalker`. This also implements
`call` instruction handling within a try-catch; every call can possibly
throw and unwind to the innermost catch block.

This adds tests for RedundantSetElimination pass, which uses
`CFGWalker`.

### CODE REVIEW

File: src/cfg/cfg-traversal.h [hunk 1]
```diff
@@ -64,8 +64,15 @@ struct CFGWalker : public ControlFlowWalker<SubType, VisitorType> {
   BasicBlock* currBasicBlock;
   // a block or loop => its branches
   std::map<Expression*, std::vector<BasicBlock*>> branches;
+  // stack of the last blocks of if conditons + the last blocks of if true
```
Comment (kripken) [inline src/cfg/cfg-traversal.h]: typo in conditions

File: src/cfg/cfg-traversal.h [hunk 2]
```diff
@@ -198,6 +205,59 @@ struct CFGWalker : public ControlFlowWalker<SubType, VisitorType> {
...
+
+  static void doStartCatch(SubType* self, Expression** currp) {
+    self->tryStack.push_back(self->currBasicBlock); // last block of try body
+    self->currBasicBlock = self->catchStack.back();
+    self->catchStack.pop_back();
+  }
+
+  static void doEndTry(SubType* self, Expression** currp) {
+    auto* last = self->currBasicBlock;
+    self->startBasicBlock(); // continuation block after try
```
Comment (kripken) [inline src/cfg/cfg-traversal.h]: maybe `try => `try-catch` to clarify it's after the entire try-catch?

File: src/cfg/cfg-traversal.h [hunk 3]
```diff
@@ -238,6 +298,32 @@ struct CFGWalker : public ControlFlowWalker<SubType, VisitorType> {
...
+        self->pushTask(SubType::scan, &curr->cast<Try>()->catchBody);
+        self->pushTask(SubType::doStartCatch, currp);
+        self->pushTask(SubType::scan, &curr->cast<Try>()->body);
+        self->pushTask(SubType::doStartTry, currp);
+        return; // don't do anything else
+      }
+      case Expression::Id::ThrowId: {
+        self->pushTask(SubType::doEndThrow, currp);
+        break;
+      }
```
Comment (kripken) [inline src/cfg/cfg-traversal.h]: these two cases can be joined, like call/callindirect?

File: src/cfg/cfg-traversal.h (hunk 1 above)
Comment (aheejin) [reply src/cfg/cfg-traversal.h]: Done

File: src/cfg/cfg-traversal.h (hunk 2 above)
Comment (aheejin) [reply src/cfg/cfg-traversal.h]: Done

File: src/cfg/cfg-traversal.h (hunk 3 above)
Comment (aheejin) [reply src/cfg/cfg-traversal.h]: Good idea!

**Pull request merged**